src/main/java/com/google/ehub/servlets/LoginServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  String userEmail;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/LoginPage.html";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      response.setContentType("text/html");
+      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+      response.getWriter().println(logoutUrl);
+    } else { // User is not Loggedin //TODO(oyins): create a "Create Account" button
+      String urlToRedirectToAfterUserLogsIn = "/ProfilePage.html";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+      response.getWriter().println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
+    }
+  }
+
+  public String getEmail() {
```

Comment:
This method should be private(or even removed) since there is no need to access the email that was added outside of the servlet. Most of the values you obtain inside the Get request will most likely go back to the client side with the HttpServletResponse so I don't really see a need for this method.

Reply:
I had that method there before I started integrating the Users API. Will remove